tt_metal/impl/dispatch/kernels/cq_prefetch.cpp

```diff
@@ -184,7 +197,10 @@ void fetch_q_get_cmds(uint32_t& fence, uint32_t& cmd_ptr, uint32_t& pcie_read_pt
     }
 
     bool cmd_ready = (cmd_ptr != fence);
-    uint32_t fetch_size = (uint32_t)*prefetch_q_rd_ptr << prefetch_q_log_minsize;
+
+    uint32_t fetch_size = ((uint32_t)(*prefetch_q_rd_ptr) & ~prefetch_q_msb_mask) << prefetch_q_log_minsize;
+    bool stall_flag = (*prefetch_q_rd_ptr & prefetch_q_msb_mask) != 0;
+    stall_state = stall_flag ? STALL_NEXT : stall_state; // NOT_STALLED -> STALL_NEXT
```

Comment:
If we move the stall check into before we read the fetchq, then we could get the stall_flag bit and shift it into position such that we don't need the branch.  That is, use STALL_NEXT=2, STALLED=1, NOT_STALLED=0.  Shift the flag so that if set we are STALL_NEXT, always shift right to get to STALLED and clear w/ an assignment

Reply:
This makes sense.. updates enums, and using this for handling of stall_flag as per suggestion:

```
    bool stall_flag = (prefetch_q_rd_ptr_local & prefetch_q_msb_mask) != 0;
    stall_state = stall_flag << 1; // NOT_STALLED -> STALL_NEXT if stall_flag is set
```

However, don't see a way ariound this branch on STALL_NEXT handling...

```
        read_from_pcie<preamble_size>
            (prefetch_q_rd_ptr, pending_read_size, fence, pcie_read_ptr, cmd_ptr, fetch_size);
        if (stall_state == STALL_NEXT) {
            barrier_and_stall(pending_read_size, fence); // STALL_NEXT -> STALLED
        }

```

... so might as well use asignment instead of cryptic right shift I figure to reach `STALLED`, inside this function (code moved out of read_from_pcie()) :

```
// If prefetcher must stall after this fetch, wait for data to come back, and move to stalled state.
FORCE_INLINE
void barrier_and_stall(uint32_t& pending_read_size, uint32_t& fence) {
    noc_async_read_barrier();
    fence += pending_read_size;
    pending_read_size = 0;
    stall_state = STALLED;
}
```